features/0699-push-notifications/README.md

```diff
@@ -0,0 +1,211 @@
+  "@id": "<UUID>",
+  "device_token": "<EXPO_TOKEN>",
+  "device_vendor": "<DEVICE_VENDOR>"
+}
+```
+
+#### Set Firebase Cloud Messaging Device Info
+
+Message to set the device info for using the [Firebase Cloud Messaging Service](https://firebase.google.com/docs/cloud-messaging) (FCM) and SDKs. Android uses FCM by default, so this service is specifically targeted at using FCM for all platforms. If this message is supported, this means the implementation supports the `fcm-notification-sender` role.
+
+```jsonc
+{
+  "@type": "https://didcomm.org/push-notifications/1.0/set-fcm-device-info",
+  "@id": "<UUID>",
+  "device_token": "<FCM_TOKEN>",
+  "device_vendor": "<DEVICE_VENDOR>"
+}
+```
+
+Description of the fields:
+
+- `device_token` -- The token that is required by the notification provider
+- `device_vendor` -- The vendor for the device (string, 'android', 'ios')
+
+It is important to note that the set device info message can be used to set, update and remove the device info. To set, and update, these values the normal messages as stated above can be used. To remove yourself from receiving push notifications, you can send the same message where all values MUST be `null`. If either value is `null` a `problem-report` can be sent back with `missing-value`.
+
+#### Get Device Info
+
+When a notification-receiver wants to get their push-notification configuration, they can send the following message:
+
+```jsonc
+{
+  "@type": "https://didcomm.org/push-notifications/1.0/get-device-info",
+  "@id": "<UUID>"
+}
+```
+
+#### Device Info
+
+Response to the get device info:
+
+```jsonc
+{
+  "@type": "https://didcomm.org/push-notifications/1.0/device-info",
+  "device_token": "<DEVICE_TOKEN>",
+  "device_vendor": "<DEVICE_VENDOR>",
+  "service": "<SERVICE>"
+  "~thread": {
+      "thid": "<GET_DEVICE_INFO_UUID>",
+  }
+}
+```
+
+- `service` -- The service used to send push notifications (string, 'native', 'expo', 'fcm')
+
+This message can be used by the notification-receiver to receive their device info, e.g., `device_token` and `device_vendor`. If the notification-sender does not have these fields for that connection, a `problem-report` can be used as a response with `not-registered-for-push-notifications`.
+
+#### Adopted messages
+
+In addition, the [`ack`](https://github.com/hyperledger/aries-rfcs/blob/08653f21a489bf4717b54e4d7fd2d0bdfe6b4d1a/features/0015-acks/README.md) message is adopted into the protocol for confirmation by the notification-sender. The ack message SHOULD be sent in response to any of the set-device-info messages.
+
+### Sending Push Notifications
+
+When an agent wants to send a push notification to another agent, the payload of the push notifications MUST include the `@type` property to indicate the message is sent by the notification-sender. Guidelines on notification messages are not defined.
+
+```jsonc
+{
+  "@type": "https://didcomm.org/push-notifications",
+  ...
+}
```

Comment:
Should we include a message_tag (or message_tag_list) in the push notification to indicate which messages it is about.